src/data_structures/suffix_array.rs

```diff
@@ -22,6 +22,10 @@ use alphabets::{Alphabet, RankTransform};
 use data_structures::smallints::SmallInts;
 use data_structures::bwt::{DerefBWT, DerefLess, DerefOcc};
 
+// TODO: figure out how to make generic on `SuffixArray` type w/o `as` cast
```

Comment:
Alternatively, I could just make the relevant functions take `&SuffixArray` which (apart from dynamic dispatch) would look identical except for an `sa.as_ref()` call instead of `sa.clone()`.